flake.nix

```diff
@@ -0,0 +1,45 @@
+{
+  description = "Application packaged using poetry2nix";
+
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+  inputs.poetry2nix = {
+    url = "github:nix-community/poetry2nix";
+    inputs.nixpkgs.follows = "nixpkgs";
+  };
+
+  outputs = { self, nixpkgs, flake-utils, poetry2nix }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        # see https://github.com/nix-community/poetry2nix/tree/master#api for more functions and examples.
+        inherit (poetry2nix.legacyPackages.${system}) mkPoetryApplication;
+        pkgs = nixpkgs.legacyPackages.${system};
+
+      in
+      {
+        packages = {
+          myapp = mkPoetryApplication {
+            projectDir = self;
+            nativeBuildInputs = [
+              pkgs.python3.pkgs.cffi
+            ];
+            propagatedBuildInputs = with pkgs; [
+              gpgme
+              pkgs.gpgme.dev
+              pkgs.python3.pkgs.cffi
+            ];
+            overrides = poetry2nix.legacyPackages.${system}.overrides.withDefaults (self: super: {
+              gpg = super.gpgme;
+              notmuch2 = pkgs.python3.pkgs.notmuch2;
+            });
+
+          };
+          default = self.packages.${system}.myapp;
+        };
+
+        devShells.default = pkgs.mkShell {
+          packages = [ poetry2nix.packages.${system}.poetry ];
```

Comment:
How is this supposed to be used? I tried 
~~~sh
nix develop github:teto/alot/flake
poetry run alot
~~~
which results in an import error (configobj not found). But when I run
~~~
nix develop github:teto/alot/flake
poetry install
~~~
it fails to install the gpg and the notmuch2 bindings because it can not find the gpgme and the notmuch development packages. Should these be added here?